stdlib/public/core/StringStorageBridge.swift

```diff
@@ -197,16 +197,21 @@ extension __StringStorage {
   final internal func _fastCStringContents(
     _ requiresNulTermination: Int8
   ) -> UnsafePointer<CChar>? {
-    if isASCII {
-      return unsafe start._asCChar
+    if 0 == requiresNulTermination || asString._guts._object.isFastZeroTerminated {
+      if isASCII {
+        return unsafe start._asCChar
+      }
     }
     return nil
   }
 
   @objc(UTF8String)
   @_effects(readonly)
   final internal func _utf8String() -> UnsafePointer<UInt8>? {
-    return unsafe start
+    if asString._guts._object.isFastZeroTerminated {
```

Comment:
If we'd happened to make a non-terminated shared string and bridge it to objc before it would have been incorrect here

Reply:
Do we have a test case that exercises this?

Reply:
At the risk of sounding very "my dog ate my homework", I did write a test for this (it's not, afaict, exercisable without the new SPI), and my machine kernel panicked and I just realized that either the panic or update-checkout somehow deleted that diff. So I guess I get to rewrite it now.

Reply:
Ok, got that sorted out!